src/index.ts

```diff
@@ -1 +1,326 @@
+    retrace: { path: (v) => `cmdline-tools/${v}/bin/retrace`, package: 'cmdline-tools' },
+    avdmanager: { path: (v) => `cmdline-tools/${v}/bin/avdmanager`, package: 'cmdline-tools' },
+    lint: { path: (v) => `cmdline-tools/${v}/bin/lint`, package: 'cmdline-tools' },
+    sdkmanager: { path: (v) => `cmdline-tools/${v}/bin/sdkmanager`, package: 'cmdline-tools' },
+} as const satisfies Record<string, { path: (version: string) => string; package: string }>;
+/* eslint-enable camelcase */
+
+/** The name of an Android development tool, either unversioned or versioned. */
+export type AndroidToolName = keyof typeof versionedAndroidTools | keyof typeof unversionedAndroidTools;
+/**
+ * An Android development tool, specified as either a name (which will cause the latest installed or available version
+ * to be used) or a name with an explicit version (only possible for versioned tools).
+ *
+ * The version in this context is _not_ the version of the tool itself, but rather the version of the package that
+ * contains the tool.
+ */
+export type AndroidTool = AndroidToolName | { tool: keyof typeof versionedAndroidTools; packageVersion: string };
+
+const platform = process.platform;
+if (platform !== 'win32' && platform !== 'darwin' && platform !== 'linux')
+    throw new Error(`Unsupported platform: "${platform}"`);
+
+const ensureSdkmanager = async () => {
+    const androidHome = await globalCacheDir('andromatic');
+
+    // Create license files just like Debian and Ubuntu do, see:
+    // https://github.com/tweaselORG/meta/issues/26#issuecomment-1535036229
+    const licensesDir = join(androidHome, 'licenses');
+    await fs.mkdir(licensesDir, { recursive: true });
+    const licenseHashes = {
+        'android-googletv-license': '601085b94cd77f0b54ff86406957099ebe79c4d6',
+        'android-sdk-arm-dbt-license': '859f317696f67ef3d7f30a50a5560e7834b43903',
+        'android-sdk-license': '24333f8a63b6825ea9c5514f83c2829b004d1fee',
+        'android-sdk-preview-license': '84831b9409646a918e30573bab4c9c91346d8abd',
+        'google-gdk-license': '33b6a2b64607f11b759f320ef9dff4ae5c47d97a',
+        'mips-android-sysimage-license': 'e9acab5b5fbb560a72cfaecce8946896ff6aab9d',
+    };
+    for (const [license, hash] of Object.entries(licenseHashes)) {
+        const licenseFile = join(licensesDir, license);
+        await fs.writeFile(licenseFile, hash);
+    }
+
+    const cmdlineToolsDir = join(androidHome, 'cmdline-tools');
+    const sdkmanager = join(cmdlineToolsDir, 'latest', 'bin', 'sdkmanager');
+
+    // Download cmdline-tools to get sdkmanager.
+    if (!(await fs.exists(sdkmanager))) {
+        const cmdlineToolsUrls = {
+            win32: 'https://dl.google.com/android/repository/commandlinetools-win-9477386_latest.zip',
+            darwin: 'https://dl.google.com/android/repository/commandlinetools-mac-9477386_latest.zip',
+            linux: 'https://dl.google.com/android/repository/commandlinetools-linux-9477386_latest.zip',
```

Comment:
This hardcodes a version, doesn’t it?

Reply:
I am not sure. The `latest` sounds like it doesn't but that number might also be a version… *shrug*

Reply:
Seems like the number does change over time: https://web.archive.org/web/*/https://dl.google.com/android/repository/commandlinetools-*